tests/test_router.py

```diff
@@ -215,6 +216,10 @@ def test_side_effect_exception():
     with pytest.raises(httpcore.NetworkError):
         router.resolve(request)
 
+    request = httpx.Request("GET", "https://hot.dog")
+    with pytest.raises(TypeError):
+        router.resolve(request)
```

Comment:
Please move this from the the exception side effect test to a separate test case, e.g. something like `test_resolve_invalid_route_return_value`.